### PR TITLE
Minor compilation fixes for gcc 4.7

### PR DESCRIPTION
**Pull request by arntanguy**
This MR adds some minor modifications to allow compiling with GCC 4.7 on Debian (on the HRP2 robot).

### CODE REVIEW

File: src/Tasks/QPConstr.h [hunk 1]
```diff
@@ -320,6 +320,14 @@ class TASKS_DLLAPI CollisionConstr : public ConstraintFunction<Inequality>
              double ds,
              double damping,
              double dampingOff);
+    CollData(CollData &&) = default;
+    CollData(const CollData &) = delete;
+    CollData & operator=(const CollData &) = delete;
+    CollData & operator=(const CollData &&)
```
Comment (gergondet) [inline src/Tasks/QPConstr.h]: Why not default?

Comment (arntanguy) [reply src/Tasks/QPConstr.h]: For some reason that I can't figure out, ` CollData & operator=(const CollData &&) = default;` does not compile:
```
In file included from /home/hrp2user/jrl/Tasks/src/QPConstr.cpp:6:0:
/home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h:327:15: error: defaulted declaration ‘tasks::qp::CollisionConstr::CollData& tasks::qp::CollisionConstr::CollData::operator=(const tasks::qp::CollisionConstr::CollData&&)’
/home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h:327:15: error: does not match expected signature ‘tasks::qp::CollisionConstr::CollData& tasks::qp::CollisionConstr::CollData::operator=(tasks::qp::CollisionConstr::CollData&&)’
/home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h: In instantiation of ‘static _OI std::__copy_move<true, false, std::random_access_iterator_tag>::__copy_m(_II, _II, _OI) [with _II = tasks::qp::CollisionConstr::CollData*; _OI = tasks::qp::CollisionConstr::CollData*]’:
/usr/include/c++/4.7/bits/stl_algobase.h:384:70:   required from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = true; _II = tasks::qp::CollisionConstr::CollData*; _OI = tasks::qp::CollisionConstr::CollData*]’
/usr/include/c++/4.7/bits/stl_algobase.h:422:39:   required from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = true; _II = __gnu_cxx::__normal_iterator<tasks::qp::CollisionConstr::CollData*, std::vector<tasks::qp::CollisionConstr::CollData> >; _OI = __gnu_cxx::__normal_iterator<tasks::qp::CollisionConstr::CollData*, std::vector<tasks::qp::CollisionConstr::CollData> >]’
/usr/include/c++/4.7/bits/stl_algobase.h:486:47:   required from ‘_OI std::move(_II, _II, _OI) [with _II = __gnu_cxx::__normal_iterator<tasks::qp::CollisionConstr::CollData*, std::vector<tasks::qp::CollisionConstr::CollData> >; _OI = __gnu_cxx::__normal_iterator<tasks::qp::CollisionConstr::CollData*, std::vector<tasks::qp::CollisionConstr::CollData> >]’
/usr/include/c++/4.7/bits/vector.tcc:139:2:   required from ‘std::vector<_Tp, _Alloc>::iterator std::vector<_Tp, _Alloc>::erase(std::vector<_Tp, _Alloc>::iterator) [with _Tp = tasks::qp::CollisionConstr::CollData; _Alloc = std::allocator<tasks::qp::CollisionConstr::CollData>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<tasks::qp::CollisionConstr::CollData*, std::vector<tasks::qp::CollisionConstr::CollData> >; typename std::_Vector_base<_Tp, _Alloc>::pointer = tasks::qp::CollisionConstr::CollData*]’
/home/hrp2user/jrl/Tasks/src/QPConstr.cpp:334:22:   required from here
/home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h:307:10: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = sch::CD_Pair; _Dp = std::default_delete<sch::CD_Pair>; std::unique_ptr<_Tp, _Dp> = std::unique_ptr<sch::CD_Pair>]’
In file included from /usr/include/c++/4.7/memory:86:0,
                 from /home/hrp2user/jrl/Tasks/src/Tasks/QPSolver.h:9,
                 from /home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h:20,
                 from /home/hrp2user/jrl/Tasks/src/QPConstr.cpp:6:
/usr/include/c++/4.7/bits/unique_ptr.h:263:19: error: declared here
In file included from /usr/include/c++/4.7/bits/char_traits.h:41:0,
                 from /usr/include/c++/4.7/ios:41,
                 from /usr/include/c++/4.7/istream:40,
                 from /usr/include/c++/4.7/sstream:39,
                 from /usr/include/c++/4.7/complex:47,
                 from /usr/local/include/eigen3/Eigen/Core:28,
                 from /home/hrp2user/jrl/Tasks/src/Tasks/QPConstr.h:9,
                 from /home/hrp2user/jrl/Tasks/src/QPConstr.cpp:6:
/usr/include/c++/4.7/bits/stl_algobase.h:348:8: note: synthesized method ‘tasks::qp::CollisionConstr::CollData& tasks::qp::CollisionConstr::CollData::operator=(const tasks::qp::CollisionConstr::CollData&&)’ first required here 
```

but ` CollData & operator=(const CollData &&) { return *this; }` does.

Comment (aescande) [reply src/Tasks/QPConstr.h]: Shouldn't you move the data before returning `this` ? This might compile, but is semantically incorrect.

Comment (gergondet) [reply src/Tasks/QPConstr.h]: I missed it yesterday but the argument shouldn’t be const, the default operator should be fine then

Comment (arntanguy) [reply src/Tasks/QPConstr.h]: Thanks for the review, this indeed compiles as expected. 
Sorry for the mistake, I am definitely not used enough to move semantics ;)

**Pull request merged**